src/plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -217,7 +217,10 @@ define(function (require) {
             if (rows == null && oldRows == null) return status.LOADING;
 
             var rowsEmpty = _.isEmpty(rows);
-            if (rowsEmpty && fetchStatus) return status.LOADING;
+            // an undefined fetchStatus means the requests are still being
+            // prepared to be sent
+            if (_.isUndefined(fetchStatus)) return status.LOADING;
+            else if (rowsEmpty && fetchStatus) return status.LOADING;
```

Comment:
If I remember correctly, `fetchStatus` will be `null` once all of the segmented requests have been received.  It'll be `undefined` up until the moment the overall wrapper for segmented search requests is marked as started.  I'm fine with moving it to a variable if it adds clarity.